cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -58,6 +61,9 @@ var (
 	argDomain              = flag.String("domain", "cluster.local", "domain under which to create names")
 	argEtcdMutationTimeout = flag.Duration("etcd-mutation-timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
 	argEtcdServer          = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
+	argEtcdCAFile          = flag.String("etcd-cafile", "", "Certificate Authority file to secure etcd communication")
+	argEtcdCertFile        = flag.String("etcd-certfile", "", "TLS certificate file to secure etcd communication")
+	argEtcdKeyFile         = flag.String("etcd-keyfile", "", "TLS key file to secure etcd communication")
```

Comment:
Ah I see. I haven't looked carefully yet. Feel free to refactor if it makes
your life easier!

On Tue, Mar 22, 2016 at 1:51 PM, a-palchikov <notifications@github.com>
wrote:

> In cluster/addons/dns/kube2sky/kube2sky.go
> <https://github.com/kubernetes/kubernetes/pull/23331#discussion_r57066568>
> :
>
> > @@ -58,6 +61,9 @@ var (
> >  	argDomain              = flag.String("domain", "cluster.local", "domain under which to create names")
> >  	argEtcdMutationTimeout = flag.Duration("etcd-mutation-timeout", 10*time.Second, "crash after retrying etcd mutation for a specified duration")
> >  	argEtcdServer          = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
> > +	argEtcdCAFile          = flag.String("etcd-cafile", "", "Certificate Authority file to secure etcd communication")
> > +	argEtcdCertFile        = flag.String("etcd-certfile", "", "TLS certificate file to secure etcd communication")
> > +	argEtcdKeyFile         = flag.String("etcd-keyfile", "", "TLS key file to secure etcd communication")
>
> I'm already using/augmenting the existing storage.etcd.EtcdConfig - it's
> just that the config itself does not expose much. Functionality-wise, it
> would be great if it exposed newClient which I'm basically duplicating
> (together with newTransport)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23331/files/48e7e271b7b4463daba3bdd90b2ea7a029e2d9a7#r57066568>
>